src/main/java/io/airlift/units/ThreadCount.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.units;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.OptionalInt;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import static io.airlift.units.Preconditions.checkArgument;
+import static java.lang.Math.min;
+import static java.lang.Math.round;
+import static java.lang.Math.toIntExact;
+import static java.lang.String.format;
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.lines;
+
+public class ThreadCount
+        implements Comparable<ThreadCount>
+{
+    private static final String PER_CORE_SUFFIX = "C";
+    private static final Supplier<Integer> AVAILABLE_PROCESSORS = MachineInfo::getAvailablePhysicalProcessorCount;
+    private final int threadCount;
+
+    ThreadCount(int threadCount)
+    {
+        checkArgument(threadCount >= 0, "Thread count cannot be negative");
+        this.threadCount = threadCount;
+    }
+
+    public int getThreadCount()
+    {
+        return threadCount;
+    }
+
+    public ThreadCount nextPowerOfTwo()
```

Comment:
It will be harder to use this in Trino (see example that @sopel39 shared): ```taskConcurrency = min(max(nextPowerOfTwo(getAvailablePhysicalProcessorCount()), 2), 32);```

Reply:
> It will be harder to use this in Trino (see example that @sopel39 shared): `taskConcurrency = min(max(nextPowerOfTwo(getAvailablePhysicalProcessorCount()), 2), 32);`

This seems to be more of a complain about the min/max than the power of two call.  I'd switch to Guava's `Ints.constrainToRange` so this code would become:
```java
taskConcurrency = constrainToRange(nextPowerOfTwo(getAvailablePhysicalProcessorCount()), 2, 32)
```
Which I think is totally readable.  BTW, Java 21 has [Math.clamp](https://bugs.openjdk.org/browse/JDK-8301226).

Reply:
I'm fine with either keeping this method or dropping it.

Reply:
I've dropped nextPowerOfTwo